end_to_end/end_to_end_suite_test.go

```diff
@@ -2657,4 +2657,175 @@ LANGUAGE plpgsql NO SQL;`)
+			testhelper.AssertQueryRuns(backupConn, `
+					CREATE TABLE part_c PARTITION OF t_part_1_prt_1 FOR VALUES IN ('c');
+					CREATE TABLE part_d PARTITION OF t_part_1_prt_1 FOR VALUES IN ('d');
+
+					INSERT INTO part_c SELECT a, 0, a, 'c' FROM generate_series(1, 10)a;
+					INSERT INTO part_d SELECT a, 0, a, 'd' FROM generate_series(1, 20)a;`)
+
+			timestamp := gpbackup(gpbackupPath, backupHelperPath, "--backup-dir", backupDir, "--leaf-partition-data",
+				"--include-table", "public.part_d")
+
+			metadataFileContents := string(getMetdataFileContents(backupDir, timestamp, "metadata.sql"))
+			Expect(metadataFileContents).ToNot(ContainSubstring("part_c"))
+			Expect(metadataFileContents).To(ContainSubstring("part_d"))
+
+			assertArtifactsCleaned(restoreConn, timestamp)
+		})
+
+		It("ignore partitions specified in exclude-table during restore", func() {
+			testhelper.AssertQueryRuns(backupConn, `
+					CREATE TABLE part_c PARTITION OF t_part_1_prt_1 FOR VALUES IN ('c');
+					CREATE TABLE part_d PARTITION OF t_part_1_prt_1 FOR VALUES IN ('d');
+
+					INSERT INTO part_c SELECT a, 0, a, 'c' FROM generate_series(1, 10)a;
+					INSERT INTO part_d SELECT a, 0, a, 'd' FROM generate_series(1, 20)a;`)
+
+			timestamp := gpbackup(gpbackupPath, backupHelperPath, "--backup-dir", backupDir, "--leaf-partition-data")
+
+			//to prevent new partitions from being created during restoration, we need to exclude the root.
+			gprestore(gprestorePath, restoreHelperPath, timestamp, "--backup-dir", backupDir, "--redirect-db", "restoredb",
+				"--exclude-table", "public.t_part")
+			assertTablesNotRestored(restoreConn, []string{"part_c", "part_d"})
+
+			assertArtifactsCleaned(restoreConn, timestamp)
+		})
+
+		It("allows the user to exclude partitions that interfere with recovery if the default partition in the extension contains data", func() {
+			testhelper.AssertQueryRuns(backupConn, `
+					ALTER EXTENSION test_ext6 DROP TABLE d_part_1_prt_extra;
+					ALTER TABLE d_part SPLIT DEFAULT PARTITION START (10) END (20);
+					ALTER EXTENSION test_ext6 ADD TABLE d_part_1_prt_extra;`)
+
+			timestamp := gpbackup(gpbackupPath, backupHelperPath, "--backup-dir", backupDir, "--leaf-partition-data")
+			gprestore(gprestorePath, restoreHelperPath, timestamp, "--backup-dir", backupDir, "--redirect-db", "restoredb",
+				"--exclude-table", "public.d_part")
+
+			assertArtifactsCleaned(restoreConn, timestamp)
+		})
+
+		It("does not lose inheritance during backup if the parent table is in the extension", func() {
+			testhelper.AssertQueryRuns(backupConn, `
+				CREATE TABLE child (d int) INHERITS (parent);
+			`)
+
+			defer testhelper.AssertQueryRuns(backupConn, `DROP TABLE child;`)
+
+			timestamp := gpbackup(gpbackupPath, backupHelperPath, "--backup-dir", backupDir)
+			metadataFileContents := string(getMetdataFileContents(backupDir, timestamp, "metadata.sql"))
+			Expect(metadataFileContents).To(ContainSubstring("INHERITS (public.parent)"))
+
```

Comment:
Here we can do restore too and check inheritance in catalog tables.

Reply:
fixed